contracts/pool/BundleService.sol

```diff
@@ -262,6 +266,60 @@ contract BundleService is
         instance.getBundleSet().unlinkPolicy(policyNftId);
     }
 
+    /// @inheritdoc IBundleService
+    function withdrawBundleFees(NftId bundleNftId, Amount amount) 
+        public 
+        virtual
+        // TODO: restricted() (once #462 is done)
+        returns (Amount withdrawnAmount) 
+    {
+        (NftId poolNftId,, IInstance instance) = _getAndVerifyActiveComponent(POOL());
+        InstanceReader reader = instance.getInstanceReader();
+        
+        IComponents.ComponentInfo memory poolInfo = reader.getComponentInfo(poolNftId);
+        address poolWallet = poolInfo.wallet;
+        
+        IBundle.BundleInfo memory bundleInfo = reader.getBundleInfo(bundleNftId);
+        
+        // determine withdrawn amount
+        withdrawnAmount = amount;
+        if (withdrawnAmount.gte(AmountLib.max())) {
+            withdrawnAmount = reader.getFeeAmount(bundleNftId);
+        } else {
+            if (withdrawnAmount.gt(reader.getFeeAmount(bundleNftId))) {
+                revert ErrorBundleServiceFeesWithdrawAmountExceedsLimit(withdrawnAmount, reader.getFeeAmount(bundleNftId));
+            }
+        }
+
+        if (withdrawnAmount.eqz()) {
+            revert ErrorBundleServiceFeesWithdrawAmountIsZero();
+        }
+
+        // check allowance
```

Comment:
do we have a ticket that says that we want/need to centralize this functionality? maybe into a lib?

Reply:
[#417](https://github.com/etherisc/gif-next/issues/417)